test/sequential/rolldown.withdraw.preOperation.test.ts

```diff
@@ -0,0 +1,578 @@
+    expect(sequencersList.toHuman().Ethereum).toContain(event.assignee);
+    expect(event.source).toEqual("PeriodReached");
+    expect(event.range.from.toNumber()).toEqual(
+      nextRequestIdEth + (batchSize - 1),
+    );
+    expect(event.range.to.toNumber()).toEqual(
+      nextRequestIdEth + (batchSize - 1),
+    );
+  });
+
+  test("GIVEN <batchSize - 1> withdraws and create manualBatch And wait for a timed generation WHEN create batch with <batchSize - 1> withdrawal and add another one THEN a batch is generated automatically", async () => {
+    let event: any;
+    await Sudo.batchAsSudoFinalized(
+      ...(await Rolldown.createABatchWithWithdrawals(
+        testUser,
+        gaspIdL1Asset.ethereum,
+        batchSize - 1,
+      )),
+    );
+    await signTx(
+      getApi(),
+      await Rolldown.createManualBatch("EthAnvil"),
+      testUser.keyRingPair,
+    ).then((events) => {
+      const res = getEventResultFromMangataTx(events);
+      expect(res.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+    });
+    event = await Rolldown.waitForNextBatchCreated(
+      "Ethereum",
+      waitingBatchPeriod,
+    );
+    expect(event.assignee).toEqual(testUser.keyRingPair.address);
+    expect(event.source).toEqual("Manual");
+    const nextRequestId = (await Rolldown.getL2RequestsBatchLast()).rangeTo + 1;
+    await Sudo.batchAsSudoFinalized(
+      ...(await Rolldown.createABatchWithWithdrawals(
+        testUser,
+        gaspIdL1Asset.ethereum,
+        batchSize - 1,
+      )),
+    );
+    await signTx(
+      getApi(),
+      await Withdraw(testUser, BN_HUNDRED, gaspIdL1Asset.ethereum, "Ethereum"),
+      testUser.keyRingPair,
+    ).then((events) => {
+      const res = getEventResultFromMangataTx(events);
+      expect(res.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+    });
+    event = await Rolldown.waitForNextBatchCreated(
+      "Ethereum",
+      waitingBatchPeriod,
+    );
+    const sequencersList = await SequencerStaking.activeSequencers();
+    expect(sequencersList.toHuman().Ethereum).toContain(event.assignee);
+    expect(event.source).toEqual("AutomaticSizeReached");
+    expect(event.range.from.toNumber()).toEqual(nextRequestId);
+    expect(event.range.to.toNumber()).toEqual(nextRequestId + (batchSize - 1));
+  });
+
+  test("GIVEN a manually generated batch, from A,B, WHEN timed happen THEN automatic batch will skip those manual ids", async () => {
+    await Sudo.batchAsSudoFinalized(
+      ...(await Rolldown.createABatchWithWithdrawals(
+        testUser,
+        gaspIdL1Asset.ethereum,
+        batchSize - 1,
+      )),
+    );
+    await signTx(
+      getApi(),
+      await Rolldown.createManualBatch("EthAnvil"),
+      testUser.keyRingPair,
+    ).then((events) => {
+      const res = getEventResultFromMangataTx(events);
+      expect(res.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+    });
+    const eventManually = await Rolldown.waitForNextBatchCreated(
```

Comment:
this event must be a result of the signTx, no need to wait. Plz fix at least only here, so we ensure that for manual batches, the event is triggered in the same block.

Reply:
Done